src/components/Header/Header.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useEffect, RefObject } from 'react';
+import { Todo } from '../../types/TodoProps';
+
+interface HeaderProps {
+  todos: Todo[];
+  query: string;
+  setQuery: (value: string) => void;
+  onFormSubmit: (event: React.FormEvent) => void;
+  onToggleAll: () => void;
+  isAdding: boolean;
+  inputRef: RefObject<HTMLInputElement>;
+  onFocusInput: () => void;
+}
+
+export const Header: React.FC<HeaderProps> = ({
+  todos,
+  query,
+  setQuery,
+  onFormSubmit,
+  onToggleAll,
+  isAdding,
+  inputRef,
+  onFocusInput,
+}) => {
+  const allCompleted = todos?.every(todo => todo.completed);
+
+  useEffect(() => {
+    if (!isAdding && !query) {
+      const timer = setTimeout(() => {
+        inputRef.current?.focus();
+      }, 50);
+
+      return () => clearTimeout(timer);
+    }
+
+    return;
+  }, [query, isAdding, inputRef]);
+
+  return (
+    <header className="todoapp__header">
+      <button
+        type="button"
+        className={`todoapp__toggle-all ${allCompleted ? 'active' : ''}`}
```

Comment:
To set classes conditionally use [classnames](https://www.npmjs.com/package/classnames) or [clsx](https://www.npmjs.com/package/clsx) library